base client: adding support for proxy option

enoodle opened the pull request.

abonas commented:
imo the review comments from https://github.com/abonas/kubeclient/pull/169 stand here as well
update: at least the ones regarding tests, code clarity and readme.

enoodle commented:
I added a test in `spec/unit/base_client.rb` to test that the argument is passed to the rest_client that is created. The proxy functionality itself is implemented in RestClient and I assume that is tested there. I also changed the parameter in `options` to be named `http_proxy_uri` for clarity.

pilhuhn commented on lib/hawkular/base_client.rb:
```diff
@@ -26,6 +26,7 @@ def initialize(entrypoint = nil,
         verify_ssl: OpenSSL::SSL::VERIFY_PEER,
         ssl_client_cert: nil,
         ssl_client_key: nil,
+        http_proxy_uri: nil,
```
👍
